x-pack/plugins/maps/public/shared/layers/sources/client_file_source/geojson_file_source.js

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { AbstractVectorSource } from '../vector_source';
+import React from 'react';
+import { GEOJSON_FILE } from '../../../../../common/constants';
+import { ClientFileCreateSourceEditor } from './create_client_file_source_editor';
+import { ESSearchSource } from '../es_search_source';
+import uuid from 'uuid/v4';
+import _ from 'lodash';
+
+export class GeojsonFileSource extends AbstractVectorSource {
+
+  static type = GEOJSON_FILE;
+  static title = 'Import GeoJSON Vector File';
+  static description = 'Upload a GeoJSON file & Index in Elasticsearch';
+  static icon = 'emsApp';
+  static indexReadyFile = true;
+
+  static createDescriptor(featureCollection, name) {
+    return {
+      type: GeojsonFileSource.type,
+      featureCollection,
+      name
+    };
+  }
+
+  static viewIndexedData = (addAndViewSource, inspectorAdapters, onIndexSuccess) => {
+    return ({ fields, id, success }) => {
+      if (!success) {
+        console.error('Unable to view indexed data');
+        return;
+      }
+      const geoFieldArr = fields.filter(
+        field => ['geo_point', 'geo_shape'].includes(field.type)
```

Comment:
use the declarations in `common/constants` for these field-types. There's a few other instances of this as well.